sw/device/silicon_creator/keys/README.md

```diff
@@ -0,0 +1,98 @@
+This directory contains ASN1 DER encoded development keys for signing ROM\_EXT
+images. Until we have a more ergonomic tool for working with keys and signatures,
```

Comment:
What do you envision as the more ergonomic tool?  Do you see keygen and signature verification as a capability of the signing command hierarchy in `opentitantool`? (ie: a user can use the tool to verify a firmware image independently of the verify code in the ROM).

Keygen would be nice to have too, _but_, real silicon_{creator, owner} keys will be generated by best-practice processes for creating and managing of high-value secrets (usually an offline process involving secure facilities).  There is still a purpose for having the tool generate keys: test keys for FPGA/simulation environments and/or generation of low-privilege owner keys (think an app-level testing key).

I'd suggest you write a short proposal document for what you'd want the more ergonomic tool(s) to look like and how you'd expect them to operate.

Reply:
> Do you see keygen and signature verification as a capability of the signing command hierarchy in opentitantool?

Yes. Since the tool already knows about the fields of the manifest, verifying the signature shouldn't require a lot of effort.

> Keygen would be nice to have too, but, real silicon_{creator, owner} keys will be generated by best-practice processes for creating and managing of high-value secrets (usually an offline process involving secure facilities). There is still a purpose for having the tool generate keys: test keys for FPGA/simulation environments and/or generation of low-privilege owner keys (think an app-level testing key).

I was thinking more along the lines of inspecting public keys, e.g. printing the modulus and exponent in little-endian that can be easily transformed to the structs we use (or possibly generate the struct definition) and maybe print the expected encoded message for a message (or hash).

> I'd suggest you write a short proposal document for what you'd want the more ergonomic tool(s) to look like and how you'd expect them to operate.

Sounds good.